Lib/tokenize.py

```diff
@@ -214,25 +232,29 @@ def untokenize(self, iterable):
                 startline = False
             elif tok_type == FSTRING_MIDDLE:
                 if '{' in token or '}' in token:
+                    token = self.escape_brackets(token)
+                    last_line = token.splitlines()[-1]
                     end_line, end_col = end
-                    end = (end_line, end_col + token.count('{') + token.count('}'))
-                    token = re.sub('{', '{{', token)
-                    token = re.sub('}', '}}', token)
-
+                    extra_chars = last_line.count("{{") + last_line.count("}}")
+                    end = (end_line, end_col + extra_chars)
+            elif tok_type in (STRING, FSTRING_START) and self.prev_type in (STRING, FSTRING_END):
+                self.tokens.append(" ")
```

Comment:
Does it work for the following examples?
```py
if 1:
    "foo"
    "bar"
```
```py
if 1:
    ("foo"
     "bar")
```
```py
if 1:
    "foo"
"bar"
```

Reply:
Yeah it works:

```
from io import StringIO
from tokenize import untokenize, generate_tokens
codes = [
    """
if 1:
    "foo"
"bar"
""",
    """
if 1:
    ("foo"
     "bar")
""",
    """
if 1:
    "foo"
    "bar"
"""
]

for code in codes:
    print(code == untokenize(generate_tokens(StringIO(code).readline)))
```

This prints:

```
./python check.py
True
True
True
```

Reply:
I added them to the test suite

Reply:
Perhaps they were bad examples. I noticed that `self.prev_type` is not always set and tried to construct examples in which you can get a dedent or newline token between two strings.

Reply:
I think that they are still good to cover as I checked and we don't have anything direct that covers these cases, only through the test random files. So I think is good to leave them even if they were not useful in the way you were thinking